hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+
+### Note on Sharing of CSRNG Instance State Variables
+
+Once an application interface port has received an `instantiate` command, that application interface port then has access to a unique CSRNG instance, which is shared by all endpoints on the same EDN.
+Therefore from a security perspective, an attack to that particular CSRNG instance is an attack on all the endpoints that share the same EDN.
+Meanwhile, seeds and other state variables specific to a particular CSRNG instance are not shared between endpoints on *separate* EDN instances, or with any hardware devices with direct connections to dedicated CSRNG application interface ports.
+
+## Interactions with Peripheral Devices
+
+Peripheral ports distribute data to the endpoints using four signals: `req`, `ack`, `bus`, and `fips`.
+
+The EDN is activated in request/acknowledge mode whenever both `EDN_ENABLE` and `BCAST_MODE_DIS` bits are simultaneously asserted in {{< regref "CTRL" >}}.
+In request/acknowledge mode, fresh (i.e. previously unseen) random values are distributed to the endpoints via the parametrizable-width `bus` signal, in response to an `req` signal.
+Whenever new values are placed on the `bus`, the `ack` is asserted until the values are consumed by the endpoint, as indicated by simultaneous assertion of the `req` and `ack` signals in the same cycle.
+Otherwise `ack` is deasserted until enough fresh bits are received from CSRNG.
+
+The `fips` signal is used to identify whether the values received on the `bus`, have prepared with complete adherence to the recommendations in NIST SP 800-90.
+Typically if the `fips` signal is deasserted, it means the associated CSRNG instance has been instantiated with a pre-FIPS seed.
+
+Endpoints can exercise careful control of the `req` and `ack` signals to manage how long random values stay asserted on the bus.
+For instance, if fresh random numbers are asserted on `bus`, but the endpoint deasserts `req` in the same cycle, those random values are not consumed and will be left unchanged until the endpoint asserts `req` to consume them.
+The `req` signal can be asserted even when fresh values are not required, to simply clear the bus of already used data, or to eliminate potentially stale data, meaning data which may have been sitting on the bus longer than security considerations would deem acceptable.
+When using the `req` signal to clear already consumed or stale data from the `bus`, the `ack` signal is monitored to confirm that the undesired data has been successfully discarded.
+
+Most peripheral ports support only one endpoint, except for peripheral port 0, and only when the EDN is enabled in broadcast mode.
+To enable broadcast mode, *assert* `EDN_ENABLE` and *deassert* `BCAST_MODE_DIS` in the {{< regref "CTRL" >}} register.
+When the EDN in broadcast mode, peripherals on port 0  do not use the `req` signal to request new numbers, rather they receive bursts of numbers at timed intervals
```

Comment:
I've been wondering about this a bit. Do we know of a situation where we have a device connected to the EDN that may or may want to be in broadcast mode? I would have thought we'd know how we're using port 0 when we wire everything together. If so, can't we get rid of the `BCAST_MODE_DIS` register entirely and put the logic in a parameter? That would make firmware's life easier, and would make the system much simpler, I think.

Reply:
Yeah.   This is a big nuisance with broadcast mode.  I'll just remove that mode.

Reply:
All references to broadcast mode removed in next commit.